Nos/Service/NamesAPI.swift

```diff
@@ -51,15 +51,33 @@ class NamesAPI {
         throw Error.unexpected
     }
 
+    /// Verifies that a given username is free to claim in nos.social
     func verify(username: String, keyPair: KeyPair) async throws -> Bool {
+        try await verify(
+            username: username,
+            host: verificationURL,
+            keyPair: keyPair,
+            valueWhenNotFound: true
+        )
+    }
+
+    /// Verifies that a given username_at_host NIP-05 can be connected to the keyPair
+    ///
+    /// - Parameter valueWhenNotFound: What to return if the server returns 404
+    func verify(
+        username: String,
+        host: URL,
+        keyPair: KeyPair,
+        valueWhenNotFound: Bool = false
+    ) async throws -> Bool {
```

Comment:
I think this is pretty similar to `RelayService.verifyNIP05(...)`, maybe we can get rid of that one?

Reply:
You're right, I missed that. Maybe we can use that instead?

Reply:
I removed `RelayService.verifyNIP05`.